src/lib/babel-custom.js

```diff
@@ -96,6 +96,10 @@ export default babelPlugin.custom(babelCore => {
 				preset.file.request.includes('@babel/preset-env'),
 			);
 
+			const environmentPreset = customOptions.modern
+				? '@babel/preset-modules'
+				: '@babel/preset-env';
+
```

Comment:
I think we also need to convert `@babel/preset-env` to use `environmentPreset` on L107:

https://github.com/developit/microbundle/pull/557/files#diff-de034110aceba806ff222e668bb57fc1R107